### PR TITLE
Copter: Change the immediate value to the defined name

### PR DESCRIPTION
**Pull request by muramura**
I think that the definition name is easier to understand than 1.

### CODE REVIEW

**Comment by rmackay9** [comment]
Should this be an enum maybe?

**Comment by peterbarker** [comment]
`enum class` :-)

**Comment by muramura** [comment]
@rmackay9 san, @peterbarker san.
I changed the definition to enum class.

File: ArduCopter/mode_throw.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ void ModeThrow::run()
     case Throw_Disarmed:
 
         // prevent motors from rotating before the throw is detected unless enabled by the user
-        if (g.throw_motor_start == 1) {
+        if (g.throw_motor_start == (int8_t)Parameters::THROW_MOTOR_START::RUNNING) {
```
Comment (peterbarker) [inline ArduCopter/mode_throw.cpp]:
```suggestion
        if (g.throw_motor_start == (uint8_t)Parameters::THROW_MOTOR_START::RUNNING) {
```

File: ArduCopter/mode_throw.cpp [hunk 2]
```diff
@@ -123,7 +123,7 @@ void ModeThrow::run()
     case Throw_Detecting:
 
         // prevent motors from rotating before the throw is detected unless enabled by the user
-        if (g.throw_motor_start == 1) {
+        if (g.throw_motor_start == (int8_t)Parameters::THROW_MOTOR_START::RUNNING) {
```
Comment (peterbarker) [inline ArduCopter/mode_throw.cpp]:
```suggestion
        if (g.throw_motor_start == (uint8_t)Parameters::THROW_MOTOR_START::RUNNING) {
```

**Comment by muramura** [comment]
@peterbarker san.. I understand the type. Thank you for the type change.

File: ArduCopter/Parameters.h [hunk 3]
```diff
@@ -447,6 +447,10 @@ class Parameters {
     AP_Int16        gcs_pid_mask;
 
 #if MODE_THROW_ENABLED == ENABLED
+    enum class THROW_MOTOR_START : uint8_t {
```
Comment (peterbarker) [inline ArduCopter/Parameters.h]: Argh.  Sorry, I neglected to suggest changing the class name; we tend to use mixed-caps for the enum-class name:

```suggestion
    enum class ThrowMotorStart : uint8_t {
```

File: ArduCopter/mode_throw.cpp [hunk 4]
```diff
@@ -108,7 +108,7 @@ void ModeThrow::run()
     case Throw_Disarmed:
 
         // prevent motors from rotating before the throw is detected unless enabled by the user
-        if (g.throw_motor_start == 1) {
+        if (g.throw_motor_start == (uint8_t)Parameters::THROW_MOTOR_START::RUNNING) {
```
Comment (peterbarker) [inline ArduCopter/mode_throw.cpp]:
```suggestion
        if (g.throw_motor_start == (uint8_t)Parameters::ThrowMotorStart::RUNNING) {
```

File: ArduCopter/mode_throw.cpp [hunk 5]
```diff
@@ -123,7 +123,7 @@ void ModeThrow::run()
     case Throw_Detecting:
 
         // prevent motors from rotating before the throw is detected unless enabled by the user
-        if (g.throw_motor_start == 1) {
+        if (g.throw_motor_start == (uint8_t)Parameters::THROW_MOTOR_START::RUNNING) {
```
Comment (peterbarker) [inline ArduCopter/mode_throw.cpp]:
```suggestion
        if (g.throw_motor_start == (uint8_t)Parameters::ThrowMotorStart::RUNNING) {
```

File: ArduCopter/Parameters.cpp [hunk 6]
```diff
@@ -695,7 +695,7 @@ const AP_Param::Info Copter::var_info[] = {
     // @Description: Used by THROW mode. Controls whether motors will run at the speed set by THR_MIN or will be stopped when armed and waiting for the throw.
     // @Values: 0:Stopped,1:Running
     // @User: Standard
-    GSCALAR(throw_motor_start, "THROW_MOT_START", 0),
+    GSCALAR(throw_motor_start, "THROW_MOT_START", (uint8_t)Parameters::THROW_MOTOR_START::STOPPED),
```
Comment (peterbarker) [inline ArduCopter/Parameters.cpp]:
```suggestion
    GSCALAR(throw_motor_start, "THROW_MOT_START", (uint8_t)Parameters::ThrowMotorStart::STOPPED),
```

**Comment by muramura** [comment]
@peterbarker san..
As I understood, enum class name does not use "_", it is written in capital letters.

**Comment by peterbarker** [comment]
I've just realised the commits need squashing here.

File: ArduCopter/Parameters.h [hunk 7]
```diff
@@ -447,6 +447,10 @@ class Parameters {
     AP_Int16        gcs_pid_mask;
 
 #if MODE_THROW_ENABLED == ENABLED
+    enum class ThrowMotorStart : uint8_t {
```
Comment (peterbarker) [inline ArduCopter/Parameters.h:450]: This should be within the throw mode, not in the parameters class.

**Review by peterbarker — CHANGES_REQUESTED** [review]
This should only be a single commit.

**Comment by peterbarker** [comment]
Replaced by https://github.com/ArduPilot/ardupilot/pull/15982 - thanks!

**Pull request closed without merging**